src/scanner.c

```diff
@@ -602,8 +602,7 @@ static Symbolic s_symop(wchar_vec s, State *state) {
       default: return con_or_var(c);
     }
   } else {
-    bool is_comment = true;
-    for (size_t i = 0; i < s.len; i++) { is_comment &= s.data[i] == '-'; }
+    bool is_comment = (s.data[0] == '-') & (s.data[1] == '-');
```

Comment:
I'd bet the compiler does some magic here anyways so I doubt it matters, but double ampersand prevents the RHS expression from being evaluated if the LHS is false.